File: javascript/hamming/hamming.js
export function compute(left, right) {
    if (left.length === 0 && right.length !== 0) {
        throw new Error('left strand must not be empty');
    } else if (right.length === 0 && left.length !== 0) {
        throw new Error('right strand must not be empty');
    } else if (left.length !== right.length) {
        throw new Error('left and right strands must be of equal length');
    }
    return [...Array(left.length).keys()].filter(i => left[i] !== right[i]).length;
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
    fun compute(seq1: String, seq2: String): Int {
        if (seq1.length != seq2.length) {
            throw IllegalArgumentException("left and right strands must be of equal length.")
        }
        return seq1.zip(seq2).count { it.first != it.second }
    }
}
